### PR TITLE
Updates Sensu description

### PR DESCRIPTION
**Pull request by asachs01**
Hi @brian-brazil , including you here since you're listed as a Maintainer. This doc change updates the Sensu description with more updated information.

### CODE REVIEW

**Comment by lucperkins** [comment]
@asachs01 First, you'll need to provide a [DCO](https://github.com/probot/dco#how-it-works) on your commit (and all future commits).

File: content/docs/introduction/comparison.md [hunk 1]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
 If you want to do whitebox monitoring, or have a dynamic or cloud based
 environment, then Prometheus is a good choice.
 
+From https://prometheus.io/docs/introduction/comparison/
```
Comment (lucperkins) [inline content/docs/introduction/comparison.md]: Remove this (as it's a link to the current page)

Comment (asachs01) [reply content/docs/introduction/comparison.md]: @lucperkins Yup, just saw that

File: content/docs/introduction/comparison.md [hunk 2]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 ## Prometheus vs. Sensu
 
-[Sensu](https://sensuapp.org/) is broadly speaking a more modern Nagios.
+[Sensu](https://sensu.io) is a composable monitoring pipeline that can reuse existing Nagios checks.
 
 ### Scope
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the Nagios check specification (https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions.
```
Comment (lucperkins) [inline content/docs/introduction/comparison.md]: The link here should be a standard Markdown link (as in the rest of the PR)

File: content/docs/introduction/comparison.md [hunk 3]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 ### Scope
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the Nagios check specification (https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions. 
 
-The primary difference is that Sensu clients [register themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
-and can determine the checks to run either from central or local configuration.
-Sensu does not have a limit on the amount of perfData.
+The primary difference is that Sensu clients [register themselves](https://docs.sensu.io/sensu-core/latest/reference/clients),
```
Comment (lucperkins) [inline content/docs/introduction/comparison.md]: I'd say remove the comma here

File: content/docs/introduction/comparison.md [hunk 4]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
+and can determine the checks to run either from central or local configuration. Sensu clients are also able to be automatically deregistered using a number of [plugins](https://github.com/sensu-plugins/sensu-plugins-sensu) and [integrations](https://docs.sensu.io/sensu-enterprise/latest/built-in-handlers/)
```
Comment (lucperkins) [inline content/docs/introduction/comparison.md]: "able to be" is a bit awkward. I'd say:

"Sensu clients can also be automatically deregistered..."

File: content/docs/introduction/comparison.md [hunk 5]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 
-There is also a [client socket](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-the-sensu-client-socket) permitting arbitrary check results to be pushed into Sensu.
+There is also a [client socket](https://docs.sensu.io/sensu-core/latest/reference/clients/#what-is-the-sensu-client-socket) permitting ad-hoc check results to be pushed into Sensu. The client socket can also be used as a receiver for alerts generated via applications emitting JSON formatted data, [SNMP traps](https://docs.sensu.io/sensu-core/latest/guides/snmp-sensu-guide/), or serve as a way to [create proxy (just-in-time) clients](https://docs.sensu.io/sensu-core/1.4/reference/clients/#proxy-clients).
```
Comment (lucperkins) [inline content/docs/introduction/comparison.md]: ad hoc

File: content/docs/introduction/comparison.md [hunk 6]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 
 ### Data model
 
-Sensu has the same rough data model as [Nagios](/docs/introduction/comparison/#prometheus-vs-nagios).
+Similar to Nagios, Sensu is host-based. The default [Sensu package](https://sensu.io/features/downloads) contains the [`sensu-api` process](https://docs.sensu.io/sensu-core/latest/api/overview/), which allows for programmatic interaction with Sensu. Sensu clients and check are also able make use of custom attributes, which function as tags and labels. These can be queried using the Sensu API.
```
Comment (nikkiki) [inline content/docs/introduction/comparison.md]: Sensu clients and check**s** are also able **to** make use of custom attributes...

Comment (lucperkins) [inline content/docs/introduction/comparison.md]: "Sensu clients and check are also able make use of custom attributes" -> "Sensu clients and checks can also use custom attributes"

File: content/docs/introduction/comparison.md (hunk 2 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: This doesn't help someone who doesn't already have good knowledge of Sensu. What's a subscription for example? This is meant to be a 10,000 foot view, keep it very generic.

File: content/docs/introduction/comparison.md (hunk 3 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Difference to what?

File: content/docs/introduction/comparison.md (hunk 5 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Could this be made a bit shorter?

**Comment by lucperkins** [comment]
I've made some tech-writer-y critiques. I'll leave content critique to others.

File: content/docs/introduction/comparison.md (hunk 6 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 4 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 3 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 2 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 6 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: This section is about data model, not APIs to access that data.

File: content/docs/introduction/comparison.md [hunk 7]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
+
+Some Sensu plugins (https://github.com/sensu-plugins) also perform contact routing-like functions.
 
 ### Storage
 
-Sensu has storage in Redis called stashes. These are used primarily for storing
-silences. It also stores all the clients that have registered with it.
+Sensu uses Redis to persist monitoring data, including the Sensu client registry, check results, check execution history, and current event data. By default, the last 21 check results are stored. Sensu provides a dashboard ([Uchiwa](https://uchiwa.io/#/)) for check data visualization.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Are those 21 results in redis or rabbit?  The 21 is too much detail in any case.

A dashboard is not storage.

File: content/docs/introduction/comparison.md [hunk 8]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 
-Sensu has a [number of components](https://sensuapp.org/docs/0.27/overview/architecture.html). It uses
+Sensu has a [number of components](https://docs.sensu.io/sensu-core/latest/overview/architecture/). It uses
 RabbitMQ as a transport, Redis for current state, and a separate server for
-processing.
+processing and API access. 
 
-Both RabbitMQ and Redis can be clustered. Multiple copies of the server can be
-run for scaling and redundancy.
+All components of a Sensu deployment (RabbitMQ, Redis, and Sensu Server/API) can be clustered for highly available and redundant configurations. The Sensu Server process, when in a clustered configuration, performs its own internal leader election for check request scheduling and event processing.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: No need to get into how exactly it works, that the feature exists is enough to mention.

File: content/docs/introduction/comparison.md [hunk 9]
```diff
@@ -244,43 +244,39 @@ blackbox probing is sufficient.
...
 
 ### Scope
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the Nagios check specification (https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions. 
 
-The primary difference is that Sensu clients [register themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
-and can determine the checks to run either from central or local configuration.
-Sensu does not have a limit on the amount of perfData.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Are you sure you want to remove this line?

**Comment by asachs01** [comment]
Signed-off-by: Aaron Sachs aaronm.sachs@gmail.com

File: content/docs/introduction/comparison.md (hunk 7 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 8 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: :heavy_check_mark:

File: content/docs/introduction/comparison.md (hunk 3 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: Good question. This is from the original comparison doc, so I've got no context as to what it was being compared against. I'm assuming it's Nagios? The intent for the change here was to point to the updated docs page. I can tweak this, though.

Comment (brian-brazil) [reply content/docs/introduction/comparison.md]: It was to Nagios, but you removed that from the previous sentence.

Comment (asachs01) [reply content/docs/introduction/comparison.md]: Ah. I'll add it back.

Comment (brian-brazil) [reply content/docs/introduction/comparison.md]: As long as it reads properly.

File: content/docs/introduction/comparison.md [hunk 10]
```diff
@@ -246,41 +246,36 @@ environment, then Prometheus is a good choice.
...
 ## Prometheus vs. Sensu
 
-[Sensu](https://sensuapp.org/) is broadly speaking a more modern Nagios.
+[Sensu](https://sensu.io) is a composable monitoring pipeline that can reuse existing Nagios checks.
 
 ### Scope
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the [Nagios check specification](https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Subscriptions and standalone are Sensu terminology, you want someone who knows only Nagios (or not even Nagios) to understand what this means. So say for example that they can be defined on each host, or centrally in sensu itself - which you cover in the next line. So I'd say remove everything after the link.

You should also add to Nagios that it all comes from the server.

File: content/docs/introduction/comparison.md [hunk 11]
```diff
@@ -246,41 +246,36 @@ environment, then Prometheus is a good choice.
...
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the [Nagios check specification](https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions. 
 
-The primary difference is that Sensu clients [register themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
-and can determine the checks to run either from central or local configuration.
-Sensu does not have a limit on the amount of perfData.
+The primary difference is that Sensu clients [register themselves](https://docs.sensu.io/sensu-core/latest/reference/clients)
+and can determine the checks to run either from central or local configuration. Sensu clients can also be automatically deregistered using a number of [plugins](https://github.com/sensu-plugins/sensu-plugins-sensu) and [integrations](https://docs.sensu.io/sensu-enterprise/latest/built-in-handlers/)
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: I'm not sure degegistration should be mentioned, that's kinda implied with this sort of system and we don't mention it for any of the others.

File: content/docs/introduction/comparison.md [hunk 12]
```diff
@@ -246,41 +246,36 @@ environment, then Prometheus is a good choice.
...
 ### Scope
 
-The same general scope differences as in the case of
-[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+Sensu checks adhere to the [Nagios check specification](https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/pluginapi.html). Sensu checks operate on two different models: publish and subscribe or standalone. It is possible to silence not just individual checks, but entire subscriptions. 
 
-The primary difference is that Sensu clients [register themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
-and can determine the checks to run either from central or local configuration.
-Sensu does not have a limit on the amount of perfData.
+The primary difference is that Sensu clients [register themselves](https://docs.sensu.io/sensu-core/latest/reference/clients)
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: You're not comparing it to anything to have a difference.

File: content/docs/introduction/comparison.md [hunk 13]
```diff
@@ -246,41 +246,36 @@ environment, then Prometheus is a good choice.
...
+and can determine the checks to run either from central or local configuration. Sensu clients can also be automatically deregistered using a number of [plugins](https://github.com/sensu-plugins/sensu-plugins-sensu) and [integrations](https://docs.sensu.io/sensu-enterprise/latest/built-in-handlers/)
 
-There is also a [client socket](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-the-sensu-client-socket) permitting arbitrary check results to be pushed into Sensu.
+There is also a [client socket](https://docs.sensu.io/sensu-core/latest/reference/clients/#what-is-the-sensu-client-socket) permitting ad-hoc check results to be pushed into Sensu. The client socket can also be used as a receiver for alerts generated via applications emitting JSON formatted data, [SNMP traps](https://docs.sensu.io/sensu-core/latest/guides/snmp-sensu-guide/), or serve as a way to [create proxy (just-in-time) clients](https://docs.sensu.io/sensu-core/1.4/reference/clients/#proxy-clients).
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: This is still a bit wordy. Delete ". The client socket can also be used as a receiver for alerts generated via applications emitting JSON formatted data," and replace with " such as "

File: content/docs/introduction/comparison.md [hunk 14]
```diff
@@ -246,41 +246,36 @@ environment, then Prometheus is a good choice.
...
 
 ### Data model
 
-Sensu has the same rough data model as [Nagios](/docs/introduction/comparison/#prometheus-vs-nagios).
+Similar to Nagios, Sensu is host-based. The default [Sensu package](https://sensu.io/features/downloads) contains the [`sensu-api` process](https://docs.sensu.io/sensu-core/latest/api/overview/), which allows for programmatic interaction with Sensu. Sensu clients and checks are also able make use of custom attributes, which function as tags and labels. These can be queried using the Sensu API.
```
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: This is still talking about things which are not data model.

File: content/docs/introduction/comparison.md (hunk 12 above)
Comment (asachs01) [reply content/docs/introduction/comparison.md]: So lines #253-#255 are from the original comparison doc. Nothing has changed in terms of the content. In fact, I reverted the change that I'd made. 

The original comparison was to Nagios. IMO it would seem that if the original doc wasn't providing any substantive comparison, then this bit should be completely removed. I'm happy to remove line 255 if you feel like that makes a bit more sense. If not, do you have a suggestion for something more appropriate to put here?

Comment (brian-brazil) [reply content/docs/introduction/comparison.md]: I think I might have gotten mixed up between different version of the change, it seems okay now.

**Comment by brian-brazil** [comment]
Thanks!
